crates/red_knot_python_semantic/src/types/infer.rs

```diff
@@ -5155,6 +5295,17 @@ impl<'db> TypeInferenceBuilder<'db> {
         }
     }
 
+    fn report_invalid_arguments_to_annotated(&self, subscript: &ast::ExprSubscript) {
```

Comment:
This feels like it might fit in better as a free function in diagnostic.rs, similar to https://github.com/astral-sh/ruff/blob/98ef5641707cb55c9c4d91983406c6b81d83851c/crates/red_knot_python_semantic/src/types/diagnostic.rs#L1044